src/TNetwork.cpp

```diff
@@ -10,8 +10,20 @@
 #include <boost/asio/ip/address.hpp>
 #include <boost/asio/ip/address_v4.hpp>
 #include <cstring>
+#include <boost/asio.hpp>
 
-typedef boost::asio::detail::socket_option::integer<SOL_SOCKET, SO_RCVTIMEO> rcv_timeout_option;
+using boost::asio::ip::tcp;
+
+struct rcv_timeout_custom_option {
+    rcv_timeout_custom_option(int timeout) : timeout_(timeout) {}
+    template <typename Protocol>
+    void set_option(tcp::socket& socket, boost::system::error_code& ec) const {
+        boost::asio::detail::socket_option::integer<IPPROTO_TCP, TCP_KEEPIDLE> option(timeout_ / 1000);
```

Comment:
Please link to the documentation explaining that this is a thing, I don't see how this would work. I would expect to see some mention of a timeout in the typename.